src/components/AuthStoreSetter.js

```diff
@@ -0,0 +1,28 @@
+import _ from 'lodash/fp'
+import { useEffect } from 'react'
+import { useAuth } from 'react-oidc-context'
+import { processUser, reloadAuthToken } from 'src/libs/auth'
+import { useOnMount } from 'src/libs/react-utils'
+import { authStore } from 'src/libs/state'
+
+
+const AuthStoreSetter = () => {
+  const auth = useAuth()
+  useOnMount(() => authStore.update(_.set(['authContext'], auth)))
+
+  useEffect(() => {
+    return auth.events.addAccessTokenExpiring(reloadAuthToken)
+  }, [auth])
+
+  useEffect(() => {
+    return auth.events.addUserLoaded(processUser)
+  }, [auth])
+
+  useEffect(() => {
+    return auth.events.addUserUnloaded(processUser)
+  }, [auth])
+
+  return null
```

Comment:
Why do you need an explicit return of `null`?

Reply:
I don't know-- @petesantos can you explain why there needs to be a return of null? I personally would love a comment in the code that explains this.

Reply:
OK, so this is because we aren't actually rendering anything in this bastardization of a component. Could you add a comment in the code to make this clear?